tests/test_formatting.py

```diff
@@ -79,6 +79,42 @@ def test_format_comment_separator(self):
             expected.splitlines(),
         )
 
+    def test_init_lists(self):
+        # Square brackets should have no spaces after [ and before ]
+        self.assertEqual(self._format(b"[ ];").rstrip(), b"[];")
+        self.assertEqual(self._format(b"[ 1 ];").rstrip(), b"[1];")
+        self.assertEqual(self._format(b"[ 1 , 2 ];").rstrip(), b"[1, 2];")
+
+        self.assertEqual(self._format(b"local x = {};").rstrip(), b"local x = {};")
+        self.assertEqual(
+            self._format(b'global x = { [ 1 ] = "one" };').rstrip(),
+            b'global x = {[1] = "one"};',
+        )
+
+        # Test complex nodes
+        code = textwrap.dedent(
+            """\
+            global x = [1, # Comment one
+            2 # Comment two
+            ];
+            """
+        )
```

Comment:
Nit: question of taste, but this becomes slightly more compact when putting the opening `"""` on the same line as the function call.

```suggestion
        code = textwrap.dedent("""\
            global x = [1, # Comment one
            2 # Comment two
            ];
            """)
```

Similar below.